x-pack/libbeat/common/aws/credentials.go

```diff
@@ -80,102 +79,118 @@ func InitializeAWSConfig(config ConfigAWS) (awssdk.Config, error) {
+//	with more details. If credential_profile_name is empty, then default profile is used.
+func getConfigSharedCredentialProfile(beatsConfig ConfigAWS) (awssdk.Config, error) {
+	logger := logp.NewLogger("WithSharedConfigProfile")
+
+	var options []func(*awsConfig.LoadOptions) error
+	if beatsConfig.ProfileName != "" {
+		options = append(options, awsConfig.WithSharedConfigProfile(beatsConfig.ProfileName))
 	}
 
 	// If shared_credential_file is empty, then external.LoadDefaultAWSConfig
 	// function will load AWS config from current user's home directory.
 	// Linux/OSX: "$HOME/.aws/credentials"
 	// Windows:   "%USERPROFILE%\.aws\credentials"
-	if config.SharedCredentialFile != "" {
-		options = append(options, external.WithSharedConfigFiles([]string{config.SharedCredentialFile}))
+	if beatsConfig.SharedCredentialFile != "" {
+		options = append(options, awsConfig.WithSharedConfigFiles([]string{beatsConfig.SharedCredentialFile}))
 	}
 
-	awsConfig, err := external.LoadDefaultAWSConfig(options...)
+	cfg, err := awsConfig.LoadDefaultConfig(context.TODO(), options...)
 	if err != nil {
-		return awsConfig, errors.Wrap(err, "external.LoadDefaultAWSConfig failed with shared credential profile given")
+		return cfg, fmt.Errorf("awsConfig.LoadDefaultConfig failed with shared credential profile given: [%w]", err)
 	}
 
 	// Assume IAM role if iam_role config parameter is given
-	if config.RoleArn != "" {
+	if beatsConfig.RoleArn != "" {
 		logger.Debug("Using role arn and shared credential profile for AWS credential")
-		return getRoleArn(config, awsConfig), nil
+		addStaticCredentialsProvider(beatsConfig, &cfg)
+		return cfg, nil
 	}
 
 	logger.Debug("Using shared credential profile for AWS credential")
-	return awsConfig, nil
+	return cfg, nil
 }
 
-func getRoleArn(config ConfigAWS, awsConfig awssdk.Config) awssdk.Config {
-	stsSvc := sts.New(awsConfig)
-	stsCredProvider := stscreds.NewAssumeRoleProvider(stsSvc, config.RoleArn)
-	awsConfig.Credentials = stsCredProvider
-	return awsConfig
+// addStaticCredentialsProvider adds a static credentials provider to the current AWS config by using the keys stored in Beats config
+func addStaticCredentialsProvider(beatsConfig ConfigAWS, awsConfig *awssdk.Config) {
+	staticCredentialsProvider := credentials.NewStaticCredentialsProvider(
+		beatsConfig.AccessKeyID,
+		beatsConfig.SecretAccessKey,
+		beatsConfig.SessionToken)
+
+	awsConfig.Credentials = staticCredentialsProvider
+
+	return
 }
 
-// EnrichAWSConfigWithEndpoint function enabled endpoint resolver for AWS
-// service clients when endpoint is given in config.
-func EnrichAWSConfigWithEndpoint(endpoint string, serviceName string, regionName string, awsConfig awssdk.Config) awssdk.Config {
+// EnrichAWSConfigWithEndpoint function enabled endpoint resolver for AWS service clients when endpoint is given in config.
+func EnrichAWSConfigWithEndpoint(endpoint string, serviceName string, regionName string, beatsConfig awssdk.Config) (awssdk.Config, error) {
```

Comment:
Maybe we should remove this test then? since we are creating a separate function `getEndpointResolverWithOptionsFunc` just for testing hmmm